views/pages/tasks.ejs

```diff
@@ -0,0 +1,51 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+  <%- include('../partials/head'); %>
+</head>
+
+<body class="container">
+
+  <header>
+    <%- include('../partials/header'); %>
+  </header>
+
+  <main>
+    <p>
+      <%= message %>
+    </p>
+    <div class="jumbotron">
+      <% if (tasks.length> 0) { %>
+        <table class="table" <tr>
+          <th>Name</th>
+          <th>Completed</th>
+          <th colspan="2"></th>
+          </tr>
+          <% tasks.forEach(function(task){ %>
+            <tr>
+              <td>
+                <%= task.name %>
+              </td>
+              <td>
+                <%= task.completed %>
+              </td>
+              <td><a href=<%="/tasks/edit/" + task.id %> class="btn btn-primary">Edit</a></td>
+              <td><a href=<%="/tasks/delete/" + task.id %> class="btn btn-primary">Delete</a></td>
```

Comment:
Extra challenge: can you add a button that toggles the tasks completion state in the list view? Send me a plan for how you'll implement it before you do it, should you choose to accept!

Reply:
That sounds fun!

Would something like this work? Is it the best approach to this problem?

I could add a new button before the `edit` button there that has an onclick event listener that calls a JS function, this function will do a post request passing the task `id` as a param and the opposite value of `completed` in the body.

Reply:
That will work but in the spirit of EJS, can you think of a solution that will work without any client-side JavaScript?

Reply:
I'm sorry, I'm not very familiar with SSR or EJS.
If I had to guess, maybe something similar to how we are sending the data in `editTask.ejs` ?

Reply:
> I'm sorry, I'm not very familiar with SSR or EJS.

I know, that's why it's a challenge 😬 

Take your time and try to figure it out, and let me know if you have any specific questions along the way.

Just know that you can do it with nothing but HTML on the client, and the same techniques you learned as part of this lesson; namely server templates & server-side request handling.

Also "SSR" or server side rendering is usually a term used with next.js and react, in the context of it being a successor to client-side rendering in react (and similar framework). However, what we're doing here with EJS is usually called "server-side templates." Templates is absolutely the key term. If you Google for SSR, I think you'll be on the wrong track.

Reply:
I think I got it!
And to be honest it was easier than using client side js since I could reuse the `editTask.ejs` form 😁 

<img width="1313" alt="image" src="https://user-images.githubusercontent.com/8095975/189266945-92c2efe0-35e5-4f5e-b02f-678636cc88a4.png">

Only thing is, I also had to update the controller because the value sent on the body was reading as a string:
<img width="531" alt="image" src="https://user-images.githubusercontent.com/8095975/189265505-0de46bb7-3dd8-4882-8460-0114379b5cec.png">

<img width="1004" alt="image" src="https://user-images.githubusercontent.com/8095975/189265626-8211c73d-84b3-431e-81ff-54ae26021457.png">

Reply:
YES, PERFECT! Good job. 

I'm going to try to look at the `true` vs `"true"` thing this weekend (feel free to ping me if I forget). I have a feeling that a checkbox input element instead of a text input element might do the trick, but it also might be a bunch of annoying experimentation without being able to make it work so I don't think there's much value in you diving into that.

Reply:
yeah, I think since for checkbox you can only have 2 possible values it is sent as boolean while for the input must be a string